City-of-Helsinki/parking-permits
# Update VAT-based refund calculation to work with all orders

## Description

Updates:
- Update first item unused quantity calculation to take also future order item start time into account
- Refactor Refund creation to own utility
- Use order item payment unit price for total price calculation
- Use period start and end times in renewal orders

Add tests for:
- Creating a refund with permit extension request with multiple product timeframes
- Creating multiple vat refunds with permit extension request with multiple product timeframes
- Creating a refund with permit low emission vehicle
- Creating a refund with permit vehicle change with one product timeframe
- Creating a refund with permit vehicle change with multiple product timeframes
- Creating multiple vat refunds with permit vehicle change with one product timeframe
- Creating multiple vat refunds with permit vehicle change with multiple product timeframes

## Context

[PV-871](https://helsinkisolutionoffice.atlassian.net/browse/PV-871)

## How Has This Been Tested?

Through unit tests.

## Manual Testing Instructions for Reviewers

Test refund creations with:
- permit ending
- vehicle change
- address change

across the system in Webshop and Admin UI.
Observe frontend results, use Django Admin to observe results in database and validate email contents.

[PV-871]: https://helsinkisolutionoffice.atlassian.net/browse/PV-871?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
